src/DynamoCore/PackageManager/PackageManagerClient.cs

```diff
@@ -1,251 +1,141 @@
 ﻿using System;
 using System.Collections.Generic;
-using System.Linq;
-using System.Reflection;
-using System.Security.Authentication;
 using System.Threading.Tasks;
 using Dynamo.Core;
-using Dynamo.Models;
-using Dynamo.Utilities;
 
 using Greg;
+using Greg.AuthProviders;
 using Greg.Requests;
 using Greg.Responses;
 
 namespace Dynamo.PackageManager
 {
-    public class LoginStateEventArgs : EventArgs
-    {
-        public string Text { get; set; }
-        public bool Enabled { get; set; }
-
-        public LoginStateEventArgs(string text, bool enabled)
-        {
-            Text = text;
-            Enabled = enabled;
-        }
-    }
-
-    /// <summary>
-    ///     A thin wrapper on the Greg rest client for performing IO with
-    ///     the Package Manager
-    /// </summary>
     public class PackageManagerClient
     {
-
-        #region Events
-
-        internal delegate void RequestAuthenticationHandler(PackageManagerClient sender);
-        internal event RequestAuthenticationHandler RequestAuthentication;
-        private void OnRequestAuthentication()
-        {
-            if (RequestAuthentication != null)
-            {
-                RequestAuthentication(this);
-            }
-        }
-
-        #endregion
-
         #region Properties/Fields
 
-        private readonly string rootPkgDir;
-        private readonly CustomNodeManager customNodeManager;
+        [Obsolete] internal static readonly string PackageContainsBinariesConstant =
+            "|ContainsBinaries(5C698212-A139-4DDD-8657-1BF892C79821)";
 
-        [Obsolete]
-        internal readonly static string PackageContainsBinariesConstant = "|ContainsBinaries(5C698212-A139-4DDD-8657-1BF892C79821)";
+        [Obsolete] internal static readonly string PackageContainsPythonScriptsConstant =
+            "|ContainsPythonScripts(58B25C0B-CBBE-4DDC-AC39-ECBEB8B55B10)";
 
-        [Obsolete]
-        internal readonly static string PackageContainsPythonScriptsConstant = "|ContainsPythonScripts(58B25C0B-CBBE-4DDC-AC39-ECBEB8B55B10)";
+        private readonly IGregClient _client;
```

Comment:
hard to tell from the diff if these are really new. If they are they should be in .net naming scheme, i.e. no '_'.